database/rrdset.c

```diff
@@ -629,6 +628,17 @@ RRDSET *rrdset_create_custom(
             mark_rebuild |= META_CHART_UPDATED;
         }
 
+        if (unlikely(units && st->state->old_units && strcmp(st->state->old_units, units))) {
```

Comment:
Stelio hi, should `st->state->old_units` be set somewhere, i.e. when the chart is first constructed? Otherwise the if will never work?